presto-parquet/src/main/java/com/facebook/presto/parquet/reader/MetadataReader.java

```diff
@@ -85,11 +86,11 @@ public static ParquetMetadata readFooter(FSDataInputStream inputStream, Path fil
         validateParquet(fileSize >= MAGIC.length + PARQUET_METADATA_LENGTH + MAGIC.length, "%s is not a valid Parquet File", file);
         long metadataLengthIndex = fileSize - PARQUET_METADATA_LENGTH - MAGIC.length;
 
-        inputStream.seek(metadataLengthIndex);
-        int metadataLength = readIntLittleEndian(inputStream);
+        InputStream tailStream = readFully(inputStream, metadataLengthIndex, PARQUET_METADATA_LENGTH + MAGIC.length);
```

Comment:
nit: `tailStream` -> `footerStream`

Reply:
I actually have a different thought. `footerStream` looks like containing all the footer data. But here the `tailStream` is only read last 8 bytes. It only includes the `PARQUET_METADATA_LENGTH` + `MAGIC`. How do you think about this?